src/lib/x509/x509_ext.h

```diff
@@ -560,6 +562,252 @@ class BOTAN_PUBLIC_API(3, 5) TNAuthList final : public Certificate_Extension {
       std::vector<Entry> m_tn_entries;
 };
 
+/**
+ * IP Address Blocks Extension
+ *
+ * RFC 3779 X.509 Extensions for IP Addr
+ *
+*/
+class BOTAN_PUBLIC_API(3, 7) IPAddressBlocks final : public Certificate_Extension {
+   public:
+      enum class BOTAN_PUBLIC_API(3, 7) Version : uint8_t {
+         IPv4 = 4,
+         IPv6 = 16,
+      };
+
+      template <Version V>
+      class BOTAN_PUBLIC_API(3, 7) IPAddress final {
+         public:
+            IPAddress() = default;
+            IPAddress(std::span<uint8_t> v);
+
+            std::array<uint8_t, static_cast<size_t>(V)> value() const { return m_value; }
```

Comment:
Using the enum value as the implict length is fine but this function has a lot of casts. Add

```
static constexpr size_t Length = static_cast<size_t>(V);
```

and use that